tests/auth_tests/test_templates.py

```diff
@@ -53,6 +54,23 @@ def test_PasswordResetConfirmView_valid_token(self):
         self.assertContains(response, '<title>Enter new password</title>')
         self.assertContains(response, '<h1>Enter new password</h1>')
 
+    @override_settings(AUTH_USER_MODEL='auth_tests.CustomUserWithUUIDPk')
+    def test_PasswordResetConfirmView_invalid_uid(self):
+        # https://code.djangoproject.com/ticket/29206
+        rf = RequestFactory()
+        user = CustomUserWithUUIDPk.objects.create_user('jsmith', 'jsmith@example.com', 'pass')
+        user = authenticate(username=user.username, password='pass')
+        request = rf.get('/somepath/')
+        request.user = user
+        client = PasswordResetConfirmClient()
+        default_token_generator = PasswordResetTokenGenerator()
+        token = default_token_generator.make_token(user)
+        uidb64 = urlsafe_base64_encode(str(user.pk).encode()).decode() + 'Mh17ag01'
```

Comment:
I don't think you need to actually create the user (or customise the request) here. If this line reads: 

```
        uidb64 = urlsafe_base64_encode(str('NOT_A_VALID_UUID').encode()).decode()
```

... then the code reduces to: 

```
        client = PasswordResetConfirmClient()
        default_token_generator = PasswordResetTokenGenerator()
        token = default_token_generator.make_token(self.user)
        uidb64 = urlsafe_base64_encode(str('NOT_A_VALID_UUID').encode()).decode()
        url = reverse('password_reset_confirm', kwargs={'uidb64': uidb64, 'token': token})
        response = client.get(url)
        # ... continue with asserts... 
```